### ISSUE TITLE
Pod install finds source error

### ISSUE THREAD

**Issue by jackyshan** [CocoaPods/CocoaPods#11975]
### Reproduce
the demo url is
`https://github.com/jackyshan/PodSourceTest`

### Question
```
source 'https://github.com/jackyshan/test-ios-specs'

use_frameworks!

platform :ios, '10.0'

target 'PodSourceTest_Example' do
  pod 'PodSourceTest', :path => '../'
  
  
  pod 'SnapKit', :source => 'https://cdn.cocoapods.org/'
  pod 'libpag'

end

```
- I set up private source https://github.com/jackyshan/test-ios-specs.
- I specified that the source of SnapKit is https://cdn.cocoapods.org/.
- Execute pod install

![image](https://github.com/CocoaPods/CocoaPods/assets/9671825/b2132c68-8726-42b1-805b-164f49dcbcad)

**But libpag also looks from https://cdn.cocoapods.org/**
This is wrong, libpag should be looked up from https://github.com/jackyshan/test-ios-specs

### Expect
```
Analyzing dependencies
Downloading dependencies
Installing SnapKit (5.6.0)
Installing libpag (4.1.23)
```

- SnapKit looks from cdn, libpag looks from test-ios-specs, because I did not specify source as cdn for libpag
- Because https://github.com/jackyshan/test-ios-specs only has version 4.1.23 of libpag, so the installed version of libpag should be 4.1.23

### Guess
When pod install is executed, the source of SnapKit specifies https://cdn.cocoapods.org/, and this source takes effect globally. Other sources that do not specify this source are also searched through https://cdn.cocoapods.org/, which is is not as expected

**Comment by orta** [comment]
Could it be using an existing lockfile metadata?

**Comment by jackyshan** [comment]
> Could it be using an existing lockfile metadata?

i use libpag shoulbe be 4.1.23 looked up in [test-ios-specs](https://github.com/jackyshan/test-ios-specs). but it is 4.2.90 looked up in cdn source

**Comment by jackyshan** [comment]
> Could it be using an existing lockfile metadata?

I am doing pod install for the first time and there is no lockfile.

**Comment by orta** [comment]
Yeah, I'd say this probably is a bug, you're welcome to give it a shot 👍🏻 

I'd also see if:

```ruby
source 'https://github.com/jackyshan/test-ios-specs'
source 'https://cdn.cocoapods.org/'

use_frameworks!

platform :ios, '10.0'

target 'PodSourceTest_Example' do
  pod 'PodSourceTest', :path => '../'
  
  
  pod 'SnapKit'
  pod 'libpag'

end
``` 

works also, because it should be prioritising the source lookups based on the order of their appearance

**Comment by jackyshan** [comment]
> ```ruby
> test-ios-specs
> ```

I want libpag to look in test-ios-specs, but libpag looks in cdn source, which is not what I want
